lunatrace/bsl/backend/src/utils/fs.ts

```diff
@@ -0,0 +1,30 @@
+/*
+ * Copyright by LunaSec (owned by Refinery Labs, Inc)
+ *
+ * Licensed under the Business Source License v1.1
+ * (the "License"); you may not use this file except in compliance with the
+ * License. You may obtain a copy of the License at
+ *
+ * https://github.com/lunasec-io/lunasec/blob/master/licenses/BSL-LunaTrace.txt
+ *
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+import fs, { Dirent } from 'fs';
+
+export function* walk(path: string, filter: (path: string, name: string) => boolean): IterableIterator<string> {
```

Comment:
using a generator function and yield for this seems really peculiar.  I can't figure out the reasoning, especially when combined with using readdirSync which has performance issues since that blocks the node thread.  Maybe this is why you and I disagreed about how fast node could recurse a directory :p.  This will only be able to read and recurse one directory at a time, where as doing it async will be more like log(n) or at least do a lot more in parallel.

Rather than use a generator, I would just make a normal async function that returns an array of path matches.  Inside that you can have an inner recursive async function that does the walking.  

In addition to generators, you almost never see `for of` in the wild.  .forEach is much more common